### DISCUSSION TITLE
Can't send messages due to the button being disabled

### DISCUSSION THREAD

**Discussion by lfv89** [AllYourBot/hostedgpt#363] [General]
I'm hosting on Render and, out of nowhere, am currently unable to send messages. Everything was working fine just a few days ago, and everything else (chat history, settings, etc...) seems to be working just fine as well!

<img width="292" alt="image" src="https://github.com/allyourbot/hostedgpt/assets/816490/f8b0a9f1-d0e7-4390-93fc-9e085f229ff0">

PS: This is for both ChatGPT and Claude.

**Comment by keithschacht** [comment]
@lfv89 Shoot, I did some complex refactor of the composer and it's conditionalized on a feature flag. I bet that's what's doing it.

There are two ways to send a message: clicking that button and simply pressing enter. Can you confirm that: (1) you type in text, and (2) you are not able to press enter -or- to click send on there?

**Comment by keithschacht** [comment]
Oh, nevermind. I just repro'd the issue. I'm working on fixing it right now.

**Comment by keithschacht** [comment]
@lfv89 I've fixed the issue! Sorry about the trouble, but I really appreciate you reporting it here.

I just merged a commit into main. Go to your fork in github and click the "Sync Fork" button like this:

<img width="550" alt="image" src="https://github.com/allyourbot/hostedgpt/assets/35061/c046efab-9ee6-4d51-a820-d8f766ec2c46">

But yours should have a button in that flyout which says "update" or "sync" (something like that). Click that to pull in the latest changes and Render should automatically re-deploy.

If you don't see Render automatically re-deploying (the 3 bouncing dots indicate this), then you go can into Render, click your Web Service, and click here to deploy the latest commit:

<img width="494" alt="image" src="https://github.com/allyourbot/hostedgpt/assets/35061/479ca5d8-7c8c-4457-af82-6b3799dd852a">

**Comment by keithschacht** [comment]
Oh, and lastly, since you're here, I would love to hear any feedback about this app. Not counting this incident :) has it been working well for you? What's been your motivation to self-host? And any requests you'd like to see with the app?

**Comment by lfv89** [comment]
Hi @krschacht, everything working fine again, thanks!

I'm self hosting because up to this point I was unable to find a good alternative for a service like the one provided by your project. That is, I'm not looking for a heavy ChatGPT replacement, I was just looking for a lighter alternative, and one that would enable me to use OpenAI's main competitors as well, which is exactly the case so far!
